packages/loader/container-definitions/src/runtime.ts

```diff
@@ -82,7 +82,12 @@ export interface IRuntime extends IDisposable {
      */
     processSignal(message: any, local: boolean);
 
-    createSummary(): ISummaryTree;
+    /**
+     * Create a summary. Used when attaching or serializing a detached container.
+     * @param blobRedirectTable - A table passed during the attach process. Used to support blobs in detached container
+     * by mapping IDs assigned locally while detached to IDs assigned by storage after attach.
+     */
+    createSummary(blobRedirectTable?: Map<string, string>): ISummaryTree;
```

Comment:
I'd add a bit more on expectations. I.e. container runtime can write out blobs and use such blob IDs to read them back while in detached state. Once container switched to attaching state (or rather when this call is done), container runtime is expected to use only remapped IDs to read from storage.